src/main/java/org/sagebionetworks/web/client/OneSageUtils.java

```diff
@@ -0,0 +1,62 @@
+package org.sagebionetworks.web.client;
+
+import static org.sagebionetworks.web.shared.WebConstants.ONESAGE_ACCOUNT_SETTINGS_PATH;
+
+import com.google.gwt.user.client.Window;
+import org.sagebionetworks.web.shared.WebConstants;
+
+public class OneSageUtils {
+
+  private static String getHostForOneSage() {
+    // SWC-6533: We do not want to stack hop for Prod and Staging
+    switch (Window.Location.getHostName().toLowerCase()) {
+      case "staging.synapse.org":
+        return "https://staging.accounts.synapse.org";
+      case "portal-dev.dev.sagebase.org":
+        return "https://accounts-dev.dev.sagebase.org";
+      case "localhost":
+      case "127.0.0.1":
+        return "http://" + Window.Location.getHostName() + ":3000";
+      default:
+        return "https://accounts.synapse.org";
+    }
+  }
```

Comment:
Separated `host` and `appId` logic so we can inject a path between them as necessary